src/shared/settings.ts

```diff
@@ -632,6 +632,20 @@ export class DevSettings extends Settings.define('aws.dev', devSettings) {
         return this.trappedSettings
     }
 
+    public isDevMode(): boolean {
+        // This setting takes precedence over everything.
+        // It must be removed completely from the settings to not be considered.
+        const forceDevMode: boolean | undefined = !this.isSet('forceDevMode')
+            ? undefined
+            : this.get('forceDevMode', false)
```

Comment:
Minor style nit: prefer placing `undefined` in the "whenFalse" expression:
```suggestion
        const forceDevMode: boolean | undefined = this.isSet('forceDevMode')
            ? this.get('forceDevMode', false)
            : undefined
```